File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem34_LeafSimilarTreesTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import static org.junit.jupiter.api.Assertions.assertFalse;

class Problem34_LeafSimilarTreesTest {

    private final Problem34_LeafSimilarTrees test = new Problem34_LeafSimilarTrees();

    @Test
    void leafSimilar() {
        TreeNode root1 = new TreeNode(1);
        root1.left = new TreeNode(2);
        root1.right = new TreeNode(3);

        TreeNode root2 = new TreeNode(1);
        root2.left = new TreeNode(3);
        root2.right = new TreeNode(2);

        assertFalse(test.leafSimilar(root1, root2));
    }
}

File: JavaScript/leetcode75/src/test/Problem34_LeafSimilarTrees.test.js
const leafSimilar = require("../main/Problem34_LeafSimilarTrees");

//  Definition for a binary tree node.
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

test("should return if leaf are similar for both tree", () => {
  let root1 = new TreeNode(1, 2, 3);
  root1.left = new TreeNode(2, null, null);
  root1.right = new TreeNode(3, null, null);

  let root2 = new TreeNode(1, 3, 2);
  root2.left = new TreeNode(3, null, null);
  root2.right = new TreeNode(2, null, null);

  expect(leafSimilar(root1, root2)).toBe(false);
});
